### PR TITLE
dashboardpage

### PR DESCRIPTION
**Pull request by roabram**
Add dashboard page to show all saved stocks out of the mongodb and changed the API call from the home page to the candlestick component.

### CODE REVIEW

File: samples.http [hunk 1]
```diff
@@ -11,8 +11,16 @@ GET {{baseUrl}}/api/singleStock?stock=AAPL
...
-{"symbol": "TSLA"}
+{"symbol": "VW"}
 
 ### GET Stock out of DB
 GET {{baseUrl}}/api/stocks HTTP/1.1  
-Accept: application/json
\ No newline at end of file
+Accept: application/json
+
+###Delete Stock out of DB
```
Comment (phlgr) [inline samples.http]:
```suggestion
### Delete Stock out of DB
```

File: src/app/App.modules.css [hunk 2]
```diff
@@ -0,0 +1,5 @@
+.App {
+  background-color: var(--primary-bg-color);
+  width: 100%;
+  height: 100vh;
+}
```
Comment (phlgr) [inline src/app/App.modules.css]: There is no class called App in the HTML

File: src/app/components/CandleStickChart/CandleStickChart.tsx [hunk 3]
```diff
@@ -1,19 +1,67 @@
...
+};
+
+function CandleStickChart({ stockSymbol }: CandleStickProps): JSX.Element {
+  const [stockData, setStockData] = useState<Array<number[]>>([]);
+
+  function loadData(symbol: string) {
+    fetch(`/api/singleStock?stock=${symbol}`)
+      .then((response) => response.json())
+      .then((data: StockData) => {
+        console.log(data);
```
Comment (phlgr) [inline src/app/components/CandleStickChart/CandleStickChart.tsx:28]:
```suggestion
```

File: src/app/components/CandleStickChart/CandleStickChart.tsx [hunk 4]
```diff
@@ -1,19 +1,67 @@
...
+        const transformed = data.historical.map((stock) => {
+          const date = new Date(stock.date);
+          console.log(date);
```
Comment (phlgr) [inline src/app/components/CandleStickChart/CandleStickChart.tsx]:
```suggestion
```

File: src/app/pages/dashboard/Dashboard.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+export type Historical = {
+  open: number;
+  close: number;
+  high: number;
+  low: number;
+  date: string;
+};
+
+// [Timestamp, Open, High, Low, Close]
```
Comment (phlgr) [inline src/app/pages/dashboard/Dashboard.tsx:15]:
```suggestion
```

File: src/utils/api.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+//     },
+//     body: JSON.stringify(user),
+//   });
+//   if (!response.ok) {
+//     const errorMessage = await response.text();
+//     throw errorMessage;
+//   }
+//   const result: UserType = await response.json();
+//   return result;
+// }
```
Comment (phlgr) [inline src/utils/api.ts:51]: Why is this here?

File: tsconfig.json [hunk 7]
```diff
@@ -20,5 +20,10 @@
     "allowSyntheticDefaultImports": true,
     "forceConsistentCasingInFileNames": true
   },
-  "include": ["./src/app"]
+  "include": [
+    "./src/app",
+    "src/app/utils",
+    "src/server/database.ts",
+    "src/server/users.ts"
```
Comment (phlgr) [inline tsconfig.json]:
```suggestion
    "./src/app",
```

**Review by phlgr — CHANGES_REQUESTED** [review]

File: src/utils/api.ts (hunk 6 above)
Comment (roabram) [reply src/utils/api.ts:51]: I wanted to build the user login and had started on the basis of a PR. Then I decided against it, because I had prioritized other things and simply forgot to delete this part.

**Comment by roabram** [comment]
> ```suggestion
    "./src/app",
```
 
👍

File: samples.http (hunk 1 above)
Comment (roabram) [reply samples.http]: 👌

File: src/app/components/CandleStickChart/CandleStickChart.tsx (hunk 4 above)
Comment (roabram) [reply src/app/components/CandleStickChart/CandleStickChart.tsx]: 👍
